execution_engine/src/storage/global_state/in_memory.rs

```diff
@@ -197,6 +197,24 @@ impl StateReader<Key, StoredValue> for InMemoryGlobalStateView {
     }
 }
 
+impl CommitProvider for InMemoryGlobalState {
```

Comment:
Maybe cfg(test)?  This is an interface we will want to deprecate eventually…

Reply:
`CommitProvider` trait provides a `commit` method, which is an extracted capacity of all global state implementations. Removing or marking this `cfg(test)` would make `InMemoryGlobalState` useless and stop `execution_engine_testing` from being able to reference it.